neuralpp/symbolic/expression.py

```diff
@@ -346,6 +353,23 @@ def __str__(self) -> str:
         return f"{self.function}({argument_str})"
 
 
+class Context(Expression, ABC):
+    @property
+    @abstractmethod
+    def unsatisfiable(self) -> bool:
+        pass
+
+    @property
+    def and_priority(self) -> int:
```

Comment:
This is for `Expression & Context` to use `Context`'s `__and__()` instead of `Expression`'s `__and__()` by default.